### PR TITLE
 Add validation support to the builder framework 

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: api/src/main/java/io/minio/BucketArgs.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public String region() {
   /** Base argument builder class. */
   public abstract static class Builder<B extends Builder<B, A>, A extends BucketArgs>
       extends BaseArgs.Builder<B, A> {
-    protected void validateName(String name) {
+    private static void validateName(String name) {
```
Comment (anjalshireesh) [inline api/src/main/java/io/minio/BucketArgs.java:37]: This method need not be static

File: api/src/main/java/io/minio/BucketArgs.java [hunk 2]
```diff
@@ -60,15 +60,28 @@ protected void validateName(String name) {
       }
     }
 
-    @SuppressWarnings("unchecked") // Its safe to type cast to T as T is inherited by this class
+    private static void validateRegion(String region) {
```
Comment (anjalshireesh) [inline api/src/main/java/io/minio/BucketArgs.java:63]: This method need not be static

File: api/src/main/java/io/minio/BucketArgs.java [hunk 3]
```diff
@@ -60,15 +60,28 @@ protected void validateName(String name) {
...
+      if (region != null && region.isEmpty()) {
```
Comment (anjalshireesh) [inline api/src/main/java/io/minio/BucketArgs.java:64]: This condition should be:

`if (region == null || region.isEmpty())`

Comment (balamurugana) [reply api/src/main/java/io/minio/BucketArgs.java:64]: Actually `null` region is allowed. When region is `null` we need to probe or use default region. However if region is provided, it should not be empty.

**Pull request merged**